src/main/kotlin/org/jitsi/nlj/util/ArrayCache.kt

```diff
@@ -93,7 +94,9 @@ open class ArrayCache<T>(
             }
             diff < 0 -> (head + diff) floorMod size
             else -> {
-                head = (head + diff) floorMod size
+                val newHead = (head + diff) floorMod size
+                lastIndexAdvanced(lastIndex, index)
```

Comment:
Is it safe to use the callback in the middle of the `doInsert` logic (before the item with the new index has been added to `cache`)?